pkg/p2p/kadcast/bucket.go

```diff
@@ -0,0 +1,99 @@
+package kadcast
+
+// MAX_BUCKET_PEERS represents the maximum
+//number of peers that a `Bucket` can hold.
+const MAX_BUCKET_PEERS uint8 = 25
+
+// Bucket stores peer info of the peers that are at a certain
+// distance range to the peer itself.
+type Bucket struct {
+	idLength         uint8
+	peerCount        uint8
+	totalPeersPassed uint64
+	// Should always be less than `MAX_BUCKET_PEERS`
+	entries []Peer
+	// This map keeps the order of arrivals for LRU
+	lru map[Peer]uint64
+	// This map allows us to quickly see if a Peer is
+	// included on a entries set without iterating over
+	// it.
+	lruPresent map[Peer]bool
```

Comment:
To mix them, I'll need to create a custom data type since Go does not allow mapping to tuples.

While it's true that it will save one mapping, i don't see how `findLRUPeerIndex` will be simplified.

Also, create an ordered data structure will cause more overheating on each `Bucket` access while the `lruPresent` mapping was basically added to take away this overheating and quickly see if a Peer is inside or not on the Buckets.

Reply:
Mmm, I am not so sure you need to create a custom data structure. I was thinking about an ordered hashset of `Peer`. If the order is the LRU, the `findLRUPeerIndex` can be as simple as the following:
```golang
    	return peerSet[0]```

Reply:
`findLRUPeerIndex` is used to get the index of the `Peer` inside the Bucket array. 
Since the bucket array does not keep order during insertions or deletions, this method is needed to locate the index of the `Peer`. So it will not be reduced since we need the `lruIndex` and the index of the peer inside of the `Bucket` array.

As mentioned before, one mapping can be reduced, but I'm now refactoring it and the code produced is more complex and ugly since the compiler cannot infer values from maps into structs. 

For code simplicity and readability, my suggestion is leave the code as it is now.  
What do you think?

Reply:
fair enough